### ISSUE TITLE
Crash bug

### ISSUE THREAD

**Issue by HGustavs** [HGustavs/LenaSYS#1895]
If one selects a non-textual file for the code viewer, one is rewarded with a crash, and a broken code example that requires a database update or a complete deletion to start working again.

Suggested solution, limiting the extensions to a known set such as for example the list of supported syntax highlighting languages. Or some other list of supported languages.

Loading a binary into the code veiwer pane and crashing is not the best option going forward.

**Comment by a15jorpa** [comment]
@HGustavs This protection seems to have been built into the file upload code, when trying to recreate this problem I had to rename a binary file with the .html extension to get it past the uploader. Is this sufficient or does codeviewer need its own protection?

**Comment by a15jorpa** [comment]
After discussions a file extension based system was deemed to be inadequate. A solution involving more extensive inspection of uploaded (possibly also viewed) files will be researched.

**Comment by a15jorpa** [comment]
This should be better now after #3384 has been mostly completed.

The "kind of file" system used in the filelink table of the database seems to classify svg images as code. When loading such an image as code a number of error messages appear, but the viewer does not seem to break. I will make a new issue about this table (#4016).
